File: model/Comment.java
package com.app.woofer.model;

import javax.persistence.*;
import java.time.Instant;

@Entity
@Table(name = "comments")
public class Comment {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @ManyToOne
    @JoinColumn(name = "userid", referencedColumnName = "id")
    private User user;

    @ManyToOne
    @JoinColumn(name = "postid", referencedColumnName = "id")
    private Post post;

    private int postId;
    private int userId;
    private Instant timestamp;
    private String body;
}

File: src/app/comment.ts
import { Post } from './post';
import { User } from './user';

/**
 * holds data for a comment
 */
export class Comment {
  Id: number;
  body: string;
  timestamp: Date;
  post: Post;
  user: User;

  constructor(body?: string, postId?: number, userId?: number) {
    this.body = body;
    this.post = new Post(postId);
    this.user = new User(userId);
  }

  public setUser(userId: number): void {
    this.user = new User();
    this.user.id = userId;
  }
  public setPost(postId: number): void {
    this.post = new Post(postId);
  }
}
